src/components/RewardCards/withdraw.tsx

```diff
@@ -73,54 +77,54 @@ const Balance = styled('div')`
 `
 
 interface RewardProps {
-  data: User
-  contract: Reward
+  user: User
+  reward: Reward
 }
 
 export default function WithdrawReward(props: RewardProps) {
   const { library, account } = useActiveWeb3React()
-  const [withdrawValue, setWithdrawValue] = useState('1000')
+  const [withdrawValue, setWithdrawValue] = useState<BigNumber | undefined>()
   const [estimate, setEstimate] = useState('1000')
 
-  function setPercentage(value: string, estimate: string) {
+  function setPercentage(value: BigNumber | undefined, estimate: string) {
     setEstimate(estimate)
     setWithdrawValue(value)
   }
 
   useEffect(() => {
-    setEstimate(props.data.rewardEstimate)
-    setWithdrawValue(props.data.lpDeposited)
-  }, [props.contract, props.data, library])
+    setEstimate(props.user.rewardEstimate)
+    setWithdrawValue(parseUnits(props.user.lpDeposited))
+  }, [props.reward, props.user, library])
 
   return (
     <Container>
       <Wrapper>
         <Text>Balance</Text>{' '}
         <Balance>
-          <span>{props.data.lpDeposited} </span> &nbsp;{' '}
-          <span>{props.contract.token0.symbol + '-' + props.contract.token1.symbol}</span>
+          <span>{props.user.lpDeposited} </span> &nbsp;{' '}
+          <span>{props.reward.token0.symbol + '-' + props.reward.token1.symbol}</span>
         </Balance>
       </Wrapper>
       <InputWrapper>
         <Input
           type="text"
           name="withdrawLP"
           id="withdrawal"
-          value={withdrawValue}
+          value={Number(withdrawValue).toString()}
           placeholder="0"
-          onChange={e => setWithdrawValue(e.target.value)}
+          onChange={e => setWithdrawValue(parseUnits(e.target.value))}
         />
-        <span>{props.contract.token0.symbol + '-' + props.contract.token1.symbol}</span>
+        <span>{props.reward.token0.symbol + '-' + props.reward.token1.symbol}</span>
       </InputWrapper>
-      <Percentage callBack={setPercentage} user={props.data} />
+      <Percentage callBack={setPercentage} user={props.user} />
       <EstimatedRewards estimate={estimate} />
       <ButtonPrimary
         onClick={() =>
           withdrawLP(
-            props.contract.LPToken,
+            props.reward.LPToken,
```

Comment:
can you move this logic into a useCallback function, then you can do you your null checks there